Add some error states to `InsightCard`

Twixes opened the pull request:
## Changes

This adds some missing `DashboardItem` empty/error state features to `InsightCard`, complimenting https://github.com/PostHog/posthog/pull/8496.

Twixes committed:
Add some error states to `InsightCard`

paolodamico reviewed (changes requested):
Haven't done a thorough review yet, but I noticed the error states seem to be cropped and you can't scroll to read them in their entirety.

<img width="623" alt="" src="https://user-images.githubusercontent.com/5864173/153375206-121f17a3-ea53-4225-8c30-86e191b48d95.png">


Also, the taking too long state shows two loading states, I would suggest removing the blue one with the overlay as the other one is clearer.

<img width="621" alt="" src="https://user-images.githubusercontent.com/5864173/153375179-2027ae99-32d8-468f-aaa3-1b5c9457ca46.png">

Twixes committed:
Clean up the empty states

Twixes committed:
Merge branch 'master' into insight-card-error-states

Twixes commented:
Good points, made sure that there'll no additional spinner over the long loading state.
Also, brought back handling for some funnel states (too few steps, invalid exclusion filters, malformed definition).

<img width="533" alt="Screen Shot 2022-02-15 at 11 54 51" src="https://user-images.githubusercontent.com/4550621/154053068-109e3a3d-ab52-4daf-9c80-afb4cdfd8454.png">
<img width="524" alt="Screen Shot 2022-02-15 at 12 22 33" src="https://user-images.githubusercontent.com/4550621/154053055-c4778f6c-b62c-44b1-8ece-c0dd564dc864.png">
<img width="515" alt="Screen Shot 2022-02-15 at 12 22 42" src="https://user-images.githubusercontent.com/4550621/154053064-7b8235aa-a9b2-41fb-b544-11adcf741027.png">

Twixes committed:
Fix `isDashboard` leftover props

Twixes committed:
Improve wording

paolodamico reviewed (approved):
Awesome stuff @Twixes, I think we could merge this, but some additional feedback.

1. I noticed that when we load the insight in view mode (and presumably in the saved insights page and/or dashboards) we don’t show the custom timeout/error messages, we use the toast instead (as we load the insight from the `/insights` endpoint instead of for example `/insights/trend/`)
2. I couldn’t get the your query is _taking_ too long to complete (with the loading thing) message to show up, it just went from loading eternally to the “Your query took too long to complete”

clarkus commented on frontend/src/scenes/insights/EmptyStates/EmptyStates.tsx:
```diff
@@ -221,29 +209,32 @@ export function FunnelSingleStepState(): JSX.Element {
                         target="_blank"
                         rel="noopener"
-                        className="flex-center"
+                        className={actionable ? 'flex-center' : undefined}
```
Yeah the alignment should be centered here for sure. These patterns are pretty inconsistent, but that's been a longer standing issue. We should plan a clean up task. Generally using the same text style and relying on the content and icon type to communicate the type of messaging (error, warning, upsell, etc).

Twixes committed:
Reword and center single step state

Twixes committed:
Address feedback

Twixes committed:
Merge branch 'master' into insight-card-error-states

Twixes commented:
Thanks, addressed.

As for Storybook, I added an example with the single-step funnel – the other states you can preview by using story controls (these affect the first insight on the grid).
![Zrzut ekranu 2022-02-16 o 21 14 14](https://user-images.githubusercontent.com/4550621/154349355-c2741fe7-f0a2-49d3-8847-65e00be6d803.png)
![Zrzut ekranu 2022-02-16 o 21 00 06](https://user-images.githubusercontent.com/4550621/154348860-931faa97-0771-40f8-8331-0ac9f0fbf731.png)

Hm, I also see the timeout message not working. Seems like `dashboardLogic` is wired up very differently from `insightLogic` so turns out that's pretty complex to add… I think we can add that in a future PR but I'd rather not put in that effort now, as I think there are more valuable things to be done.

Twixes committed:
Merge branch 'master' into insight-card-error-states

Twixes committed:
Remove useless `/i/TEST1234` test
